File: ProjectLion_SpringBoot/mission/mission3/Refactoring/refactoring_board/src/main/java/dev/yoon/refactoring_board/controller/PostController.java
package dev.yoon.refactoring_board.controller;

import dev.yoon.refactoring_board.dto.PostDto;
import dev.yoon.refactoring_board.service.PostService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;

@RestController
@Slf4j
@RequestMapping("board/{boardId}/post")
@RequiredArgsConstructor
public class PostController {

    private final PostService postService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public ResponseEntity<PostDto.Res> createPost(
            @PathVariable("boardId") Long id,
            @RequestBody @Valid PostDto.Req postDto
    ) {
        return ResponseEntity.ok(this.postService.createPost(id, postDto));

    }

    @GetMapping()
    public ResponseEntity<List<PostDto.Res>> readPostAll(
            @PathVariable("boardId") Long boardId
    ) {
        List<PostDto.Res> postDtoList = this.postService.readPostAllbyBoardId(boardId);

        return ResponseEntity.ok(postDtoList);

    }

    @GetMapping("{postId}")
    public ResponseEntity<PostDto.Res> readPostOne(
            @PathVariable("boardId") Long boardId,
            @PathVariable("postId") Long postId) {

        PostDto.Res postDto = this.postService.readPostOneByBoardId(boardId, postId);

        return ResponseEntity.ok(postDto);
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @PutMapping("{postId}")
    public ResponseEntity<?> updatePost(
            @PathVariable("boardId") Long boardId,
            @PathVariable("postId") Long postId,
            @RequestBody PostDto.Req postDto) {
        postService.updatePost(boardId, postId, postDto);

        return ResponseEntity.noContent().build();
    }

    @ResponseStatus(HttpStatus.ACCEPTED)
    @DeleteMapping("{postId}")
    public ResponseEntity<?> deletePost(
            @PathVariable("boardId") Long boardId,
            @PathVariable("postId") Long postId,
            @RequestBody PostDto.Req postDto) {
        this.postService.deletePost(boardId, postId, postDto);
        return ResponseEntity.noContent().build();

    }
}

File: developer's-note-of-incorrect-answer-for-adding-tests/kotlin-test-code-with-architecture/src/main/kotlin/com/example/kotlintestcodewitharchitecture/controller/PostController.kt
package com.example.kotlintestcodewitharchitecture.controller

import com.example.kotlintestcodewitharchitecture.model.dto.GetPostResponse
import com.example.kotlintestcodewitharchitecture.model.dto.PostUpdateDto
import com.example.kotlintestcodewitharchitecture.service.PostService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/posts")
class PostController(
    private val postService: PostService,
) {
    @GetMapping("/{id}")
    fun getPostById(@PathVariable id: Long): ResponseEntity<GetPostResponse> {
        return ResponseEntity
            .ok()
            .body(GetPostResponse.of(postService.getPostById(id)))
    }

    @PutMapping("/{id}")
    fun updatePost(@PathVariable id: Long, @RequestBody postUpdateDto: PostUpdateDto): ResponseEntity<GetPostResponse> {
        return ResponseEntity
            .ok()
            .body(GetPostResponse.of(postService.update(id, postUpdateDto)))
    }

}